docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,275 @@
+
+### Accompanying code example
+
+[The accompanying Substreams module](#) associated with this documentation is responsible for extracting a handful of data fields from the Block object injected into the Rust-based map module. The sink tool processes the extracted blockchain data line-by-line and outputs the data to the files-based persistence mechanism you've chosen.
+
+The accompanying code example extracts four data points from the Block object and packs them into the `substreams.sink.files.v1` protobuf's data model. The data is passed to the protobuf as a single line of plain text, a `CSV` entry, or a `JSONL` element. StreamingFast is working on support for binary formats such as [Avro](https://avro.apache.org/), [Parquet](https://parquet.apache.org/), and others.
+
+## Outline
+
+1. Download code and tools
+2. Run code and verify output
+3. Code walkthrough
+4. Understanding the sink tool
+5. Run tool and verify output
+6. Conclusion and review
+7. Getting help
+
+## Download code and tools
+
+### Install `substreams-sink-files`
+
+**TODO:** <i>Where is the actual binary file? What is the file type/format/extension (if any)?</i>
+https://github.com/streamingfast/substreams-sink-files/releases
+
+### Verify installation
+
+Checking the version of `substreams-sink-files` will produce a message similar to:
+
+```bash
+substreams-sink-files version dev
+```
+
+### Update system `PATH`
+
+If you want to use the `substreams-sink-tool` from anywhere on your system make sure the computer's `PATH` environment variable has been set correctly.
+
+Add the following lines to the computer's `~/.bashrc` configuration file and then restart the shell session.
+
+```bash
+export GOPATH=$HOME/go
+export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+```
+
+### Download accompanying code example
+
+The accompanying code example is available in its [official GitHub repository](#). You'll need to download or clone the repository to your computer to complete this tutorial.
+
+You can use the `git clone` command to acquire the example code.
+
+```bash
+git clone TODO_INSERT_REPO_LINK_WHEN_AVAILABLE
+```
+
+## Run code and verify output
+
+It's a good idea to run the example code using your installation of the `substreams` CLI to make sure everything is set up and working properly.
+
+Verify the setup for the example project by using the `make build`, `make codegen` and `substreams run` commands.
+
+### Build the example project
+
+First, build the Substreams module by using the included `make` command.
+
+```bash
+make
+```
+
+### Generate the required code for the example
+
+Next, run the included `make codegen` command to create the required protobuf files.
+
+```bash
+make codegen
+```
+
```

Comment:
Belongs to section where Substreams modifications is needed

Reply:
Moved.